### PR TITLE
Student view read only bcsc information on account set up74 

### PR DESCRIPTION
**Pull request by 2011sandeepraj**
No Keycloak or login related development/test is needed. - Check
 Development - Create a Vue Single Page Form on SABC-SIMS Website with the fields mentioned above in the user story. - Check
 Parse the returned token information on the Keycloak object will done by User Story 17 and information will be stored in the state object.- Check
 Display the information in the vue form .- Check
 Nothing is being stored in the database for this user story.- Check

### CODE REVIEW

File: Developer.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# SIMS Developer Notes
+
+Student Information Management System. Post-Secondary Student Financial Aid System.
+
+##IDP TEST USERS BCSC
+1. SS4BPS204 98904 FOUR
```
Comment (andrewsignori-aot) [inline Developer.md:6]: I believe that we are not allowed to store this info here.
@abschwenker, can you please confirm it?

File: sources/packages/web/src/services/AuthService.ts [hunk 2]
```diff
@@ -24,6 +24,7 @@ export default async function(
     });
     if (keycloak.authenticated) {
       store.dispatch("auth/login", keycloak);
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/services/AuthService.ts:26]: With the introduction of "setStudentProfileData", do we still need "store.dispatch("auth/login", keycloak)"?

File: sources/packages/web/src/services/AuthService.ts [hunk 3]
```diff
@@ -24,6 +24,7 @@ export default async function(
...
+      store.dispatch("student/setStudentProfileData",keycloak);
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/services/AuthService.ts:27]: Please add a space before "keycloak".

File: sources/packages/web/src/services/AuthService.ts (hunk 2 above)
Comment (andrewsignori-aot) [inline sources/packages/web/src/services/AuthService.ts:26]: With the introduction of "setStudentProfileData", do we still need "store.dispatch("auth/login", keycloak)"?

File: sources/packages/web/src/store/index.ts [hunk 4]
```diff
@@ -1,14 +1,15 @@
...
+import { student } from "./modules/student/student";
 
 const storeOptions: StoreOptions<RootState> = {
   state: {
     version: "1.0.0"
   },
   modules: {
-    auth
-  }
+      auth,student
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/store/index.ts:11]: Please, considerer breaks the line or add a space before "student".

File: sources/packages/web/src/store/modules/auth/auth.ts [hunk 5]
```diff
@@ -4,6 +4,7 @@ import { actions } from "./actions";
 import { mutations } from "./mutations";
 import { AuthState, RootState } from "../../states";
 
+
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/store/modules/auth/auth.ts:7]: This change does not seem part of this review. Is it possible to remove it?

File: sources/packages/web/src/store/modules/auth/getters.ts [hunk 6]
```diff
@@ -2,7 +2,7 @@ import { GetterTree } from "vuex";
 import { AuthState, RootState } from "../../states";
 
 export const getters: GetterTree<AuthState, RootState> = {
-  isAuthenticated(state: AuthState): boolean {
-    return state.isAuthenticated;
-  }
+    isAuthenticated(state: AuthState): boolean {
+        return state.isAuthenticated as boolean;
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/store/modules/auth/getters.ts:6]: I believe that "state.isAuthenticated" is already as boolean. Is there a reason why this change was needed?

File: sources/packages/web/src/store/modules/auth/mutations.ts [hunk 7]
```diff
@@ -2,17 +2,10 @@ import { MutationTree } from "vuex";
...
-    state.displayName = token.displayName;
-    state.email = token.email;
-    state.emailVerified = token.email_verified;
-    state.familyName = token.family_name;
-    state.gender = token.gender;
-    state.givenName = token.given_name;
-    state.identityAssuranceLevel = token.identity_assurance_level;
-  }
+    setAuthenticated(state: AuthState, token: any){
+
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/store/modules/auth/mutations.ts:6]: Please consider cleaning the additional blank lines.

File: sources/packages/web/src/store/modules/student/student.ts [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+    profile: new StudentProfile()
+}
+
+export const student = {
+    namespaced: true,
+    state : initialState,
+   
+    mutations: {
+        SET_STUDENT_PROFILE_DATA(state:StudentState, parsedToken:any){
+            //console.log(`mutation p${parsedToken.age}`)
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/store/modules/student/student.ts:30]: Please consider removing the commented lines.

File: sources/packages/web/src/store/modules/student/student.ts [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+
+const initialState: StudentState = {
+    profile: new StudentProfile()
+}
+
+export const student = {
+    namespaced: true,
+    state : initialState,
+   
+    mutations: {
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/store/modules/student/student.ts:28]: Is there any particular reason to keep the store's components (e.g. mutations, actions, etc.) in the same file instead keep them in separate files? IMHO, as they grow, it would favor maintenance and organization, but I am open to what the team decided on this matter.

File: sources/packages/web/src/views/Login.vue [hunk 10]
```diff
@@ -33,7 +33,7 @@ export default class Login extends Vue {
   login() {
     AppConfigService.shared.authService?.login({
       idpHint: "bcsc",
-      redirectUri: `${location.origin}/`
+      redirectUri: `${location.origin}/student`
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/views/Login.vue:36]: I believe that the specific redirect to the "/student" is no longer needed.
The "created" hook on App.vue will take care of it, as per my understanding.

File: sources/packages/web/src/views/Student.vue [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+    <!--The information from Services Card includes First Name, Last Name, Middle Name, Date of Birth, 
+     verified email, and Gender as read-only-->
+      
+    <div class="p-fluid p-formgrid p-grid">
+      <div class="p-field p-col">
+          <label for="firstName">First Name</label>
+          <InputText id="firstName" v-model="studentProfile.displayName" readonly />
+      </div>
+      <div class="p-field p-col">
+          <label for="familyName">Family Name</label>
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/views/Student.vue:16]: Consider change "Family Name" to "Last Name" as described in the story.

File: sources/packages/web/src/store/modules/student/student.ts [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+            state.profile.name = parsedToken.name;
+            state.profile.age = parsedToken.age;
+            state.profile.birthdate = parsedToken.birthdate;
+            state.profile.displayName = parsedToken.displayName;
+            state.profile.email = parsedToken.email;
+            state.profile.emailVerified = parsedToken.email_verified;
+            state.profile.familyName = parsedToken.family_name;
+            state.profile.gender = parsedToken.gender;
+            state.profile.givenName = parsedToken.given_name;
+            state.profile.middleName=parsedToken.middle_name;
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/store/modules/student/student.ts:40]: I believe that we are not receiving the middle_name from BCSC and I also believe that this field will be always blank.
From BCSC with receive given_names and given_name. I believe that the only way to get the middle name would be subtracting the given_names from given_name.
For instance:
Considering the given_names as "SS4BPS Benjamin"
Considering the given_name as "SS4BPS"
the middle name would be "Benjamin".
@abschwenker, could you please confirm if it is still needed?

File: sources/packages/web/src/store/states.ts [hunk 13]
```diff
@@ -3,16 +3,10 @@ export interface RootState {
 }
 
 export class AuthState {
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/store/states.ts:5]: Do we still need this AuthState?

File: sources/packages/web/tsconfig.json [hunk 14]
```diff
@@ -11,6 +11,7 @@
     "esModuleInterop": true,
     "allowSyntheticDefaultImports": true,
     "sourceMap": true,
+
```
Comment (andrewsignori-aot) [inline sources/packages/web/tsconfig.json:14]: Please check if this change is required.

**Review by andrewsignori-aot — CHANGES_REQUESTED** [review]
Hi @2011sandeepraj, I added my comments and I believe that some of then we will need a team discussion.
I believe that the majority of my comments are minors, but we need to check closer the one related to the "Middle Name" that is being requested in the story. I believe that we are not getting this info from BCSC. I added the details about it in the comments.

**Review by abschwenker — COMMENTED** [review]
Please connect to ticket. I believe you can do manually or though keywords in PR desc.

File: Developer.md (hunk 1 above)
Comment (abschwenker) [reply Developer.md:6]: Correct, please remove

File: sources/packages/web/src/views/Login.vue (hunk 10 above)
Comment (fwpushan) [reply sources/packages/web/src/views/Login.vue:36]: Yes, app should redirect to root url and from there based on authentication state of KeyCloak App.vue will redirect them to appropriate page.

**Review by fwpushan — COMMENTED** [review]
@andrewsignori-aot did a very comprehensive review, please resolve them.

File: Developer.md (hunk 1 above)
Comment (2011sandeepraj) [reply Developer.md]: Done.

File: sources/packages/web/src/store/modules/auth/getters.ts (hunk 6 above)
Comment (2011sandeepraj) [reply sources/packages/web/src/store/modules/auth/getters.ts:6]: Typescript errored out unless this was done.

File: sources/packages/web/src/store/modules/student/student.ts (hunk 9 above)
Comment (2011sandeepraj) [reply sources/packages/web/src/store/modules/student/student.ts:28]: Sure. Usually i don't like having so many small files. if and when things do grow and this needs to be broken, we can. Also, it will help will need to have one definition of whats long enough to be broken

File: sources/packages/web/src/views/Student.vue (hunk 11 above)
Comment (2011sandeepraj) [reply sources/packages/web/src/views/Student.vue]: Done

File: sources/packages/web/tsconfig.json (hunk 14 above)
Comment (2011sandeepraj) [reply sources/packages/web/tsconfig.json]: Done

File: sources/packages/web/src/store/modules/student/student.ts (hunk 12 above)
Comment (abschwenker) [reply sources/packages/web/src/store/modules/student/student.ts:40]: I will have an answer today if we need to parse out the middle name

File: sources/packages/web/src/services/AuthService.ts (hunk 3 above)
Comment (2011sandeepraj) [reply sources/packages/web/src/services/AuthService.ts:27]: running npm run lint on the whole web code base resolved this, but i dont know why this PR view on git hub isnt showing the space

Comment (2011sandeepraj) [reply sources/packages/web/src/services/AuthService.ts:27]: running npm run lint on the whole web code base resolved this, but i dont know why this PR view on git hub isnt showing the space

File: sources/packages/web/src/services/AuthService.ts (hunk 2 above)
Comment (2011sandeepraj) [reply sources/packages/web/src/services/AuthService.ts:26]: Fixed. Removed all auth related files

File: sources/packages/web/src/store/index.ts (hunk 4 above)
Comment (2011sandeepraj) [reply sources/packages/web/src/store/index.ts:11]: removed auth related files and mentions

File: sources/packages/web/src/store/modules/auth/auth.ts (hunk 5 above)
Comment (2011sandeepraj) [reply sources/packages/web/src/store/modules/auth/auth.ts:7]: removed auth related files

File: sources/packages/web/src/store/modules/auth/mutations.ts (hunk 7 above)
Comment (2011sandeepraj) [reply sources/packages/web/src/store/modules/auth/mutations.ts:6]: Done

File: sources/packages/web/src/store/modules/student/student.ts (hunk 8 above)
Comment (2011sandeepraj) [reply sources/packages/web/src/store/modules/student/student.ts]: done

File: sources/packages/web/src/store/modules/student/student.ts (hunk 9 above)
Comment (2011sandeepraj) [reply sources/packages/web/src/store/modules/student/student.ts:27]: Kept it as is as decided, this can be done later when there are more mutations

File: sources/packages/web/src/views/Login.vue (hunk 10 above)
Comment (2011sandeepraj) [reply sources/packages/web/src/views/Login.vue]: Fixed

File: sources/packages/web/src/store/modules/student/student.ts (hunk 12 above)
Comment (2011sandeepraj) [reply sources/packages/web/src/store/modules/student/student.ts:39]: Removed the middlename as discussed with schwenker

File: sources/packages/web/src/store/states.ts (hunk 13 above)
Comment (2011sandeepraj) [reply sources/packages/web/src/store/states.ts:5]: Changed

File: sources/packages/web/src/store/modules/student/student.ts (hunk 12 above)
Comment (andrewsignori-aot) [reply sources/packages/web/src/store/modules/student/student.ts:39]: @abschwenker, can we have some comments on the story to support this change?

File: Developer.md [hunk 15]
```diff
@@ -2,7 +2,4 @@
 
 Student Information Management System. Post-Secondary Student Financial Aid System.
 
-##IDP TEST USERS BCSC
-1. SS4BPS204 98904 FOUR
-2. SS4BPS202 98902 TWO
-3. SS4BPS209 98909 NIN
+
```
Comment (andrewsignori-aot) [inline Developer.md:5]: Avoid including unnecessary files in the review.
Please do not consider this as a blocker for this review.

File: sources/packages/web/tsconfig.json [hunk 16]
```diff
@@ -10,8 +10,7 @@
     "skipLibCheck": true,
     "esModuleInterop": true,
     "allowSyntheticDefaultImports": true,
-    "sourceMap": true,
-    
+    "sourceMap": true,
```
Comment (andrewsignori-aot) [inline sources/packages/web/tsconfig.json:13]: Avoid including unnecessary files in the review.
Please do not consider this as a blocker for this review.

**Review by andrewsignori-aot — APPROVED** [review]
In general, please avoid including unnecessary files in the review.
Please do not consider this as a blocker for this review.
